src/python/pants/util/strutil_test.py

```diff
@@ -160,3 +161,112 @@ def test_bullet_list_max_elements() -> None:
+    )
+    # Test with leading backslash
+    assert (
+        softwrap(
+            """\
+                Do you believe in UFOs, astral projections, mental telepathy, ESP, clairvoyance,
+                spirit photography, telekinetic movement, full trance mediums, the Loch Ness monster
+                and the theory of Atlantis?
+
+                Ah, if there's a steady paycheck in it,
+                I'll believe anything you say.
+
+                [From
+                Ghostbusters (1984)]
+            """
+        )
+        == (
+            "Do you believe in UFOs, astral projections, mental telepathy, ESP, clairvoyance, "
+            "spirit photography, telekinetic movement, full trance mediums, the Loch Ness monster "
+            "and the theory of Atlantis?"
+            "\n\n"
+            "Ah, if there's a steady paycheck in it, I'll believe anything you say."
+            "\n\n"
+            "[From Ghostbusters (1984)]"
+        )
+    )
+    # Test without leading backslash
+    assert (
+        softwrap(
+            """
+                Do you believe in UFOs, astral projections, mental telepathy, ESP, clairvoyance,
+                spirit photography, telekinetic movement, full trance mediums, the Loch Ness monster
+                and the theory of Atlantis?
+
+                Ah, if there's a steady paycheck in it,
+                I'll believe anything you say.
+
+                [From
+                Ghostbusters (1984)]
+            """
+        )
+        == (
+            "Do you believe in UFOs, astral projections, mental telepathy, ESP, clairvoyance, "
+            "spirit photography, telekinetic movement, full trance mediums, the Loch Ness monster "
+            "and the theory of Atlantis?"
+            "\n\n"
+            "Ah, if there's a steady paycheck in it, I'll believe anything you say."
+            "\n\n"
+            "[From Ghostbusters (1984)]"
+        )
+    )
+    assert (
+        softwrap(
+            """
+                Do you
+                believe in:
+
+                    UFOs
+                    astral projections
+                    mental telepathy
+                    ...
+
+                Ah, if there's a steady paycheck in it,
+                I'll believe anything you say.
+            """
+        )
+        == (
+            "Do you believe in:"
+            "\n\n"
+            "    UFOs\n"
+            "    astral projections\n"
+            "    mental telepathy\n"
+            "    ...\n"
+            "\n"
+            "Ah, if there's a steady paycheck in it, I'll believe anything you say."
+        )
+    )
+    assert (
+        softwrap(
+            """
+                Roll Call:
+
+                    ```
+                        - Dr. Peter Venkman
+                        - Dr. Egon Spengler
+                        - Dr. Raymond Stantz
+                        - Winston Zeddemore
+                    ```
+
+                All here.
+            """
```

Comment:
Can you please add one like this but with multiple lines for the list element because it's so long? 

```
Some reasons this feature is better:

1. You can do super cool things with Pants that you could not normally
 do because yay Pants
2. Blah blah
```

I think that we want all of the first entry to show up in the same "paragraph".

Reply:
I'm not sure that's technically feasible without some hardcore parsing. We might be able to swing it, but I'd prefer to punt if we could. Would you consider that a dealbreaker?

Reply:
One way to handle this semi-easily would involve replacing all instances of multiple spaces with a single space in between two word characters. If we did that, we could have the user use a trailing backslash:

```python
"""
    And not really a ghostbuster, but we need to test wrapped indentation
    - Louis (Vinz, Vinz Clortho, Keymaster of Gozer. Volguus Zildrohar, Lord of\
    the Sebouillia)
"""
```

(note the trailing slash in the middle line. It'll result in the tabbing in the middle of the word, which if we are allowed to assume comes only from this circumstance, we can strip)

Reply:
Yeah it's either the above or using an f-string with the value already softwrapped and inserting it that way.

Reply:
Here's a few examples of double-spaces that'd be stripped (not a bad thing, just an obseration)
https://github.com/pantsbuild/pants/blob/80c7849a7ccd5006fa5d88f14b9253195361f50a/build-support/bin/generate_docs.py#L188
https://github.com/pantsbuild/pants/blob/80c7849a7ccd5006fa5d88f14b9253195361f50a/src/python/pants/backend/codegen/thrift/apache/subsystem.py#L31

(Those are the only 2 help ones)

Reply:
I do use this style of lists quite frequently, so I do care about it. But, not enough to block what is an improvement 95% of the time.

I suggest we think about how to reduce risk for formatting not being correct with multiline list elements. A simple one is code review, I'm already pretty obnoxious with this 😛 Maybe update the docstring here? Linting would be totally overkill.

Reply:
Added backslash-at-the-end-of-line support